Add timer mode

GuillaumeMorini opened the pull request:
Hello,

I really appreciate your tool, but for me it was missing a timer feature, so I add it.
If you are ok to add it to your repo it would be perfect.

Thanks
Guillaume

Commit:
Add timer mode

vimist commented:
Hey Guillaume, thanks for the PR, I'll give it a read through as soon as I can :)

vimist committed:
Refactored timer mode

Also added more flexibility to modes to allow them to manage their own command
line parameters.

vimist commented:
Hey @GuillaumeMorini, thanks again for the pull request. 

I've refactored it a bit to get help make the framework a bit more flexible, but hopefully it still satisfies your requirements. I also renamed it to `countdown` instead of `timer`, this was just to reduce any confusion as to whether the timer counts up or down.

I'll give you a couple of days to get back to me and if I've not heard anything I'll merge the changes though. :)

GuillaumeMorini commented:
Perfect for me 
Thanks for taking time to integrate my change
